src/gromacs/nblib/gmxsetup.cpp

```diff
@@ -198,6 +198,31 @@ void NbvSetupUtil::unpackTopologyToGmx()
     }
 }
 
+void NbvSetupUtil::setAtomProperties(std::unique_ptr<nonbonded_verlet_t>& nbv)
+{
+    t_mdatoms mdatoms;
+    // We only use (read) the atom type and charge from mdatoms
+    mdatoms.typeA = const_cast<int*>(system_->topology().getParticleTypeIdOfAllParticles().data());
+    mdatoms.chargeA = const_cast<real*>(system_->topology().getCharges().data());
+    nbv->setAtomProperties(mdatoms, particleInfoAllVdw_);
+}
+
+void NbvSetupUtil::setParticlesOnGrid(std::unique_ptr<nonbonded_verlet_t>& nbv)
+{
+    matrix box_;
+    gmx::fillLegacyMatrix(system_->box().matrix(), box_);
```

Comment:
This is also needed by gmxcalculator, suggesting that this should be computed at a higher level. This is more of a todo suggestion than a change request.